### PR TITLE
WIP Fix i310 Removed a single DB ID restrictions

### PR DESCRIPTION
**Pull request by sign60intel**

### CODE REVIEW

**Comment by fexolm** [comment]
Could you please rebase this pr to appropriate branch, so we can see only relevant changes?

File: ArrowStorage/ArrowStorage.cpp [hunk 1]
```diff
@@ -228,9 +228,15 @@ Fragmenter_Namespace::TableInfo ArrowStorage::getTableMetadata(int db_id,
 const DictDescriptor* ArrowStorage::getDictMetadata(int db_id,
```
Comment (ienkovich) [inline ArrowStorage/ArrowStorage.cpp:228]: Don't add an additional method to get a dictionary, modify the existing one instead.

File: ArrowStorage/ArrowStorage.cpp [hunk 2]
```diff
@@ -228,9 +228,15 @@ Fragmenter_Namespace::TableInfo ArrowStorage::getTableMetadata(int db_id,
...
                                                     bool /*load_dict*/) {
-  CHECK_EQ(db_id, db_id_);
-  if (dicts_.count(dict_id)) {
-    return dicts_.at(dict_id).get();
+  UNREACHABLE();
+  return nullptr;
+}
+
+const DictDescriptor* ArrowStorage::getDictMetadata(int dict_id, bool /*load_dict*/) {
+  CHECK_EQ(dict_id, (dict_id & 0xffffff));
```
Comment (ienkovich) [inline ArrowStorage/ArrowStorage.cpp:236]: I don't get this check. `dict_id` is supposed to have `schema_id` in its highest byte. It doesn't have to be 0, it has to match the one stored in `db_id_`.

File: ArrowStorage/ArrowStorage.cpp [hunk 3]
```diff
@@ -228,9 +228,15 @@ Fragmenter_Namespace::TableInfo ArrowStorage::getTableMetadata(int db_id,
...
+  int _dict_id = dict_id & 0xffffff;
```
Comment (ienkovich) [inline ArrowStorage/ArrowStorage.cpp:237]: You should use full id here, but change the place where ID is assigned to dictionaries to include schema ID into it. See `ArrowStorage::createTable`

File: ArrowStorage/ArrowStorage.h [hunk 4]
```diff
@@ -54,6 +54,9 @@ class ArrowStorage : public SimpleSchemaProvider, public AbstractDataProvider {
   ArrowStorage(int schema_id, const std::string& schema_name, int db_id)
       : SimpleSchemaProvider(schema_id, schema_name), db_id_(db_id) {}
 
+  ArrowStorage(const std::string& schema_name, int dict_id)
```
Comment (ienkovich) [inline ArrowStorage/ArrowStorage.h:57]: I don't get why you might need this constructor. It makes no sense.

File: DataMgr/DataMgr.cpp [hunk 5]
```diff
@@ -616,9 +616,16 @@ Buffer_Namespace::CpuBufferMgr* DataMgr::getCpuBufferMgr() const {
 const DictDescriptor* DataMgr::getDictMetadata(int db_id,
                                                int dict_id,
                                                bool load_dict) const {
+  CHECK_EQ(dict_id, (dict_id & 0xffffff));
   return getPersistentStorageMgr()->getDictMetadata(db_id, dict_id, load_dict);
 }
 
+const DictDescriptor* DataMgr::getDictMetadata(int dict_id, bool load_dict) const {
+  int schema_id = dict_id >> 24;
+  return getPersistentStorageMgr()->getDictMetadata(
```
Comment (ienkovich) [inline DataMgr/DataMgr.cpp:625]: This part looks wrong. You should simply pass both parameters to the persistent storage manager.

File: DataMgr/PersistentStorageMgr/PersistentStorageMgr.cpp [hunk 6]
```diff
@@ -156,7 +158,22 @@ void PersistentStorageMgr::removeTableRelatedDS(const int db_id, const int table
...
+  } else {
+    auto gfm = dynamic_cast<File_Namespace::GlobalFileMgr*>(sm);
+    CHECK_NE(gfm, nullptr);
+    return sm->getDictMetadata(db_id, dict_id, load_dict);
+  }
+}
+
+const DictDescriptor* PersistentStorageMgr::getDictMetadata(int dict_id, bool load_dict) {
+  int schema_id = dict_id >> 24;
+  return getStorageMgr(schema_id)->getDictMetadata(
```
Comment (ienkovich) [inline DataMgr/PersistentStorageMgr/PersistentStorageMgr.cpp:175]: `getStorageMgr` makes the shift by itself, so you should simply pass `dict_id`.

File: DataMgr/PersistentStorageMgr/PersistentStorageMgr.cpp [hunk 7]
```diff
@@ -156,7 +158,22 @@ void PersistentStorageMgr::removeTableRelatedDS(const int db_id, const int table
...
+      schema_id, dict_id & 0xffffff, load_dict);
```
Comment (ienkovich) [inline DataMgr/PersistentStorageMgr/PersistentStorageMgr.cpp:176]: Simply forward both parameters.

File: QueryEngine/Execute.h [hunk 8]
```diff
@@ -1131,7 +1131,6 @@ class Executor {
   static const int32_t ERR_STRING_CONST_IN_RESULTSET{13};
   static const int32_t ERR_STREAMING_TOP_N_NOT_SUPPORTED_IN_RENDER_QUERY{14};
   static const int32_t ERR_SINGLE_VALUE_FOUND_MULTIPLE_VALUES{15};
-  static const int32_t ERR_GEOS{16};
```
Comment (ienkovich) [inline QueryEngine/Execute.h:1134]: Please don't mix in GEO stuff into the same PR. This should go into a separate one.

File: Tests/ArrowBasedExecuteTest.cpp [hunk 9]
```diff
@@ -378,17 +378,17 @@ const size_t g_num_rows{10};
 class ExecuteTestBase {
  public:
   static constexpr int TEST_SCHEMA_ID = 1;
-  static constexpr int TEST_DB_ID = (TEST_SCHEMA_ID << 24) + 1;
```
Comment (ienkovich) [inline Tests/ArrowBasedExecuteTest.cpp:381]: This file should not be modified for your patch, if you have to modify something here, then your patch is most probably wrong. Please remove all modifications from these tests.

File: Tests/ArrowStorageSqlTest.cpp [hunk 10]
```diff
@@ -29,7 +29,7 @@
 #include <gtest/gtest.h>
 
 constexpr int TEST_SCHEMA_ID = 1;
-constexpr int TEST_DB_ID = (TEST_SCHEMA_ID << 24) + 1;
```
Comment (ienkovich) [inline Tests/ArrowStorageSqlTest.cpp:32]: These tests shouldn't have any modifications as well.

File: Tests/ArrowStorageTest.cpp [hunk 11]
```diff
@@ -19,7 +19,7 @@
 #include <gtest/gtest.h>
 
 constexpr int TEST_SCHEMA_ID = 1;
-constexpr int TEST_DB_ID = (TEST_SCHEMA_ID << 24) + 1;
```
Comment (ienkovich) [inline Tests/ArrowStorageTest.cpp:22]: All changes in this file are also unnecessary.

File: Tests/NoCatalogRelAlgTest.cpp [hunk 12]
```diff
@@ -25,7 +25,7 @@
 #include <gtest/gtest.h>
 
 constexpr int TEST_SCHEMA_ID = 1;
-constexpr int TEST_DB_ID = (TEST_SCHEMA_ID << 24) + 1;
```
Comment (ienkovich) [inline Tests/NoCatalogRelAlgTest.cpp:28]: Same here

File: Tests/NoCatalogSqlTest.cpp [hunk 13]
```diff
@@ -31,7 +31,7 @@
 #include <gtest/gtest.h>
 
 constexpr int TEST_SCHEMA_ID = 1;
-constexpr int TEST_DB_ID = (TEST_SCHEMA_ID << 24) + 1;
```
Comment (ienkovich) [inline Tests/NoCatalogSqlTest.cpp:34]: And here.

**Review by ienkovich — CHANGES_REQUESTED** [review]

**Pull request closed without merging**